File: leetcode/1929. Concatenation of Array/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
// Time: ~O(n)
// Space: O(2n) ~> O(n)
int* getConcatenation(int* nums, int numsSize, int* returnSize)
{
    int *ret = malloc(sizeof(int) * (numsSize * 2));
    for (int i = 0; i < numsSize; i++) {
        ret[i] = nums[i];
        ret[numsSize + i] = nums[i];
    }
    *returnSize = numsSize*2;
    return ret;
}

File: leetcode/1929. Concatenation of Array/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
// Time: O(n)
// Space: O(n)
var getConcatenation = function(nums) {
    return nums.concat(nums);
};// Note: concat is O(n + m), here it double self, therefore O(2n) ~> O(n)
